torch/distributed/distributed_c10d.py

```diff
@@ -4041,6 +4060,8 @@ def new_subgroups(
             the construction of specific process groups. i.e. for the ``nccl``
             backend, ``is_high_priority_stream`` can be specified so that
             process group can pick up high priority cuda streams.
+        group_desc (str, optional): A string describing the group. Each subgroup will
+            amend subgroup id to its group_desc
```

Comment:
what is a "subgroup id"? As a user, I don't see any reference to it.
And what's the need for attaching id to a user-provided `group_desc`?

Reply:
in new_subgroups, each subgroup has an id. yea maybe just inheritting the group_desc is enough since subgroup_id is not exposed to users.

Reply:
Sounds good.